wps-cloudformation-template.yaml

```diff
@@ -753,6 +757,104 @@ Resources:
                   !Join ["", ["arn:aws:s3:::", !If [CreateEphemeralBucket, !Ref EphemeralS3Bucket, !FindInMap [EnvironmentMap, !Ref environment, BucketName]]]],
                   !Join ["", ["arn:aws:s3:::", !If [CreateEphemeralBucket, !Ref EphemeralS3Bucket, !FindInMap [EnvironmentMap, !Ref environment, BucketName]],"/*"]]
                 ]
+  ReloadLambdaCodeFunctionRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+      Policies:
+        - PolicyName: UpdateVersion
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'lambda:UpdateFunctionCode'
+                Resource: '*'
+  ReloadLambdaCodeFunction:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Code:
+        ZipFile: |
+          import os
```

Comment:
So many loops and prints ;) I pasted into an IDE and crunched it down a bit... untested, but a couple of list comprehensions removes some nesting.

```python
import os

import boto3
import cfnresponse

client = boto3.client('lambda')
responseData = {'Data': 'OK'}
function_name_suffix = os.getenv('FUNCTION_NAME_SUFFIX')
s3_bucket_suffix = os.getenv('S3_BUCKET_SUFFIX')
s3_key_suffix = os.getenv('S3_KEY_SUFFIX')


def handler(event, context):
    try:
        print('RequestType')
        print(event['RequestType'])
        if event['RequestType'] == 'Update':
            print(os.linesep.join('Environment Variable {} : {}'.format(k, v) for k, v in os.environ))

            function_vars = [var for var in os.environ if var.endswith(function_name_suffix)]

            for function_name in function_vars:
                function_name_prefix = function_name.replace(function_name_suffix, '')
                bucket_var = '{}{}'.format(function_name_prefix, s3_bucket_suffix)
                key_var = '{}{}'.format(function_name_prefix, s3_key_suffix)

                s3_bucket = os.getenv(bucket_var)
                s3_key = os.getenv(key_var)

                print('s3Bucket Key:{} Value:{}'.format(bucket_var, s3_bucket))
                print('s3Key Key:{} Value:{}'.format(key_var, s3_key))

                response = client.update_function_code(
                    FunctionName=function_name,
                    S3Bucket=s3_bucket,
                    S3Key=s3_key,
                    Publish=True
                )
                print('response: {}'.format(response))
        return cfnresponse.send(event, context, cfnresponse.SUCCESS, responseData, 'S3PutObjectFunction')
    except Exception as e:
        print(e)
        return cfnresponse.send(event, context, cfnresponse.FAILED, responseData, 'S3PutObjectFunction')
```

Reply:
Updated and tested the code